File: BackEnd_Learn/Java/citylife/src/main/java/com/leeseojune53/citylife/Repository/UserRepository.java
package com.leeseojune53.citylife.Repository;

import com.leeseojune53.citylife.Entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {
    Optional<User> findByUserId(String userId);
    List<User> findTop3ByOrderByUserIdDesc();
}

File: BackEnd_Learn/kotlin/kopring/src/main/kotlin/io/github/leeseojune53/kopring/domain/user/domain/repositories/UserRepository.kt
package io.github.leeseojune53.kopring.domain.user.domain.repositories

import io.github.leeseojune53.kopring.domain.user.domain.User
import org.springframework.data.repository.CrudRepository

interface UserRepository : CrudRepository<User, Long> {
	fun findByName(name: String): User?
}